### PR TITLE
[MRG] Randomly generated default conn_seed

### LINKED ISSUE #356: seed for net.add_connections ?
In the documentation here:

https://jonescompneurolab.github.io/hnn-core/dev/generated/hnn_core.Network.html#hnn_core.Network.add_connection

there is a `seed` argument. I am assuming this kicks in only if `probability < 1`, could we clarify this? Also, I think the default should be `None` -- that is random seed if `probability < 1` 

Also, we are missing units for the different parameters in the docstring

### PR DESCRIPTION
**Pull request by mjpelah**
closes #356 

* Changed conn_seed default to None from 3
* Updated docstring
* Untested (probably will break)

### CODE REVIEW

**Comment by jasmainak** [comment]
looks like tests did not break 😄 You should update `whats_new` once done. Was this discussed already with @rythorpe and @ntolley ?

**Comment by rythorpe** [comment]
> looks like tests did not break smile You should update `whats_new` once done. Was this discussed already with @rythorpe and @ntolley ?

Yeah, I was kinda surprised that all the tests passed. And yes, @mjpelah and I discussed this today! I suspect that none of our tests actually allow the default value (which is now `None`) to get passed to [`_drive_cell_event_times()`](https://github.com/jonescompneurolab/hnn-core/blob/bab819687e40c866c7c1130e1aad3ab86124d3fe/hnn_core/drives.py#L235) - we'll need to test this and probably adjust some down-stream code.

**Comment by mjpelah** [comment]
> > looks like tests did not break smile You should update `whats_new` once done. Was this discussed already with @rythorpe and @ntolley ?
> 
> Yeah, I was kinda surprised that all the tests passed. And yes, @mjpelah and I discussed this today! I suspect that none of our tests actually allow the default value (which is now `None`) to get passed to [`_drive_cell_event_times()`](https://github.com/jonescompneurolab/hnn-core/blob/bab819687e40c866c7c1130e1aad3ab86124d3fe/hnn_core/drives.py#L235) - we'll need to test this and probably adjust some down-stream code.

`conn-seed` is passed into `_connection_probability` where it is used as a seed to create the `rng` however, only if `probability != 1.0`. If not, it is defaulted to `3` (now changed to `None`)

If the seed is `None` (quoting from numpy.org): "then fresh, unpredictable entropy will be pulled from the OS," so the generator will produce different random numbers each time it is used. 

The only place `rng` is used is in selecting a random subset of connections to retain:
`new_connections = rng.choice(range(len(all_connections)), n_connections, replace=False)`

@rythorpe @jasmainak **Is this something we want to be different each time it is ran?**

In the meantime, I will edit the test, making probability 0.99, and see if it runs.

**Comment by mjpelah** [comment]
Changed `conn_seed` default to `None` in `_connection_probability`. 

Since `_connection_probability` is only called in `add_connection`, it will always already be defaulted. Therefore defaulting `conn_seed` again in `_connection_probability` is redundant.

It may be worth keeping though for ease of reading?

**Comment by rythorpe** [comment]
> Changed `conn_seed` default to `None` in `_connection_probability`.

This looks good. I just noticed that I got `conn_seed` and `event_seed` confused in my previous comment - good catch. Unless anyone else objects (cc @jasmainak @ntolley), I don't think we need a unit test for this.

> Since `_connection_probability` is only called in `add_connection`, it will always already be defaulted. Therefore defaulting `conn_seed` again in `_connection_probability` is redundant.
> 
> It may be worth keeping though for ease of reading?

I like it the way you have it. Go ahead and append a bullet point item to the "Changelog" section of `whats_new.rst` and don't forget to add your GH link at the end of the file!

**Comment by mjpelah** [comment]
> I like it the way you have it. Go ahead and append a bullet point item to the "Changelog" section of `whats_new.rst` and don't forget to add your GH link at the end of the file!

Done!

**Comment by rythorpe** [comment]
This PR turned out to be very straightforward. Any final thoughts or critiques @jasmainak @ntolley @chenghuzi?

File: doc/whats_new.rst [hunk 1]
```diff
@@ -135,7 +136,7 @@ Bug
 
 - Fix issue where common drives use the same parameters for all cell types, by `Nick Tolley`_
   in :gh:`350`
-
+
```
Comment (jasmainak) [inline doc/whats_new.rst:139]: nitpick, but try not to touch lines that are unrelated to the PR :)

**Comment by jasmainak** [comment]
also set the PR title to [MRG] when ready from your end

**Review by rythorpe — APPROVED** [review]
Looks good @mjpelah! Feel free to review and/or merge whenever you're ready @jasmainak.

**Comment by jasmainak** [comment]
Perfect, thanks @mjpelah ! Congratulations on your first ever contribution to `hnn-core` :-)
